staging/src/k8s.io/apiserver/pkg/storage/etcd3/feature/feature_support_checker_test.go

```diff
@@ -0,0 +1,249 @@
+		featureName                   string
+		mockEndpointsVersion          []mockEndpointVersion
+		mockSecondaryEndpointsVersion []mockEndpointVersion
+		expectedResult                bool
+		expectedSecondaryResult       bool
+		expectedError                 error
+		expectedCheckClientError      error
+	}{
+		{
+			testName:       "Default disabled",
+			featureName:    storage.RequestWatchProgress,
+			expectedResult: false,
+		},
+		{
+			testName:                 "Disabled with error unknown feature",
+			expectedResult:           false,
+			expectedError:            fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", ""),
+			expectedCheckClientError: fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", ""),
+		},
+		{
+			testName:    "Disables if called with one client doesn't support it",
+			featureName: storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.5.13", Endpoint: "localhost:2390"},
+				{Version: "3.5.10", Endpoint: "localhost:2391"}},
+			expectedResult: false,
+		},
+		{
+			testName:    "Disables if called with all client doesn't support it",
+			featureName: storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.5.9", Endpoint: "localhost:2390"},
+				{Version: "3.5.10", Endpoint: "localhost:2391"}},
+			expectedResult: false,
+		},
+		{
+			testName:    "Enabled if provided client has at least one endpoint that supports it and no client that doesn't",
+			featureName: storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.4.31", Endpoint: "localhost:2389"},
+				{Version: "3.5.13", Endpoint: "localhost:2390"},
+				{Version: "3.5.14", Endpoint: "localhost:2390"},
+				{Version: "3.6.0", Endpoint: "localhost:2392"}},
+			expectedResult: true,
+		},
+		{
+			testName:    "Disabled cannot be re-enabled",
+			featureName: storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.4.0", Endpoint: "localhost:3390"}},
+			mockSecondaryEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.6.0", Endpoint: "localhost:3391"}},
+			expectedResult:          false,
+			expectedSecondaryResult: false,
+		},
+		{
+			testName:    "Enabled after with one client that supports it, and later disabled it with second client that doesn't",
+			featureName: storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.6.0", Endpoint: "localhost:2390"}},
+			mockSecondaryEndpointsVersion: []mockEndpointVersion{
+				{Version: "3.4.0", Endpoint: "localhost:2391"}},
+			expectedResult:          true,
+			expectedSecondaryResult: false,
+		},
+		{
+			testName:             "Disabled if client doesn't have any endpoints",
+			featureName:          storage.RequestWatchProgress,
+			mockEndpointsVersion: []mockEndpointVersion{},
+			expectedResult:       false,
+		},
```

Comment:
Can you add one more case, error on first, success on second?

Reply:
Thanks for responding, yes i'll add more tests, i was just wondering if the test table is good.

Reply:
Proposed to create a hierarchy to the test case to make it more readable.